### PR TITLE
init: oauth docs

### PR DESCRIPTION
**Pull request by ericpaulsen**
adds new docs on how to configure GitHub OAuth.

### CODE REVIEW

File: docs/install/oauth.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# GitHub OAuth
+
+By default, Coder is accessible via built-in authentication. Alternatively, you
+can configure Coder to enable logging in through GitHub OAuth. See below for
+configuration steps.
+
+## Step 1: Configure the OAuth application in GitHub
+
+When [https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/](creating an OAuth app),
+GitHub will ask you for the following Coder parameters:
```
Comment (bpmct) [inline docs/install/oauth.md:10]:
```suggestion
First, [https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/](register a GitHub OAuth app). GitHub will ask you for the following Coder parameters:
```

File: docs/install/oauth.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Navigate to your Coder host and run the following command to start up the Coder
+server:
+
+```console
+coder server --oauth2-github-allow-signups=true --oauth2-github-allowed-orgs="your-org" --oauth2-github-client-id="8d1...e05" --oauth2-github-client-secret="57ebc9...02c24c"
+```
+
+Alternatively, if you already have a running Coder instance, you can achieve the
+same result as the command above by adding the following environment variables
+to the `/etc/coder.d/coder.env` file:
```
Comment (bpmct) [inline docs/install/oauth.md:29]:
```suggestion
Alternatively, if you are running Coder as a system servide, you can achieve the
same result as the command above by adding the following environment variables
to the `/etc/coder.d/coder.env` file:
```

File: docs/install/oauth.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+```console
+CODER_OAUTH2_GITHUB_ALLOW_SIGNUPS=true
+CODER_OAUTH2_GITHUB_ALLOWED_ORGS="your-org"
+CODER_OAUTH2_GITHUB_CLIENT_ID="8d1...e05"
+CODER_OAUTH2_GITHUB_CLIENT_SECRET="57ebc9...02c24c"
+```
+
+Once complete, run `sudo systemctl enable --now coder` to reboot Coder.
```
Comment (bpmct) [inline docs/install/oauth.md:38]:
```suggestion
Once complete, run `sudo service coder restart` to restart Coder.
```

File: docs/manifest.json [hunk 4]
```diff
@@ -26,8 +26,13 @@
         {
           "title": "Postgres",
           "description": "Learn how to create and use your own Postgres database.",
-          "path": "./postgres.md"
-        }
+          "path": "./install/postgres.md"
+        },
+        {
+          "title": "GitHub OAuth",
+          "description": "Learn how to authenticate with GitHub.",
```
Comment (bpmct) [inline docs/manifest.json:33]:
```suggestion
          "description": "Learn how to set up OAuth using your GitHub organization.",
```

File: docs/install/oauth.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# GitHub OAuth
+
+By default, Coder is accessible via built-in authentication. Alternatively, you
+can configure Coder to enable logging in through GitHub OAuth. See below for
```
Comment (bpmct) [inline docs/install/oauth.md:4]: Does this work with GitHub Enterprise or just GitHub? Might be nice to specify ;) 

cc @kylecarbs

**Review by bpmct — APPROVED** [review]

**Review by khorne3 — APPROVED** [review]

**Pull request merged**